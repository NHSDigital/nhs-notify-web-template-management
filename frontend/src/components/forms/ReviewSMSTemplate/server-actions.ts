import {
  SMSTemplate,
  TemplateFormState,
} from 'nhs-notify-web-template-management-utils';
import { redirect, RedirectType } from 'next/navigation';
import { z } from 'zod';

export const $FormSchema = z.object({
  reviewSMSTemplateAction: z.enum(['sms-edit', 'sms-submit'], {
    message: 'Select an option',
  }),
});

export async function reviewSmsTemplateAction(
  formState: TemplateFormState<SMSTemplate>,
  formData: FormData
) {
  const form = Object.fromEntries(formData.entries());

  const { success, error, data } = $FormSchema.safeParse(form);

  if (!success) {
    return {
      ...formState,
      validationError: error.flatten(),
    };
  }

  if (data.reviewSMSTemplateAction === 'sms-edit') {
    return redirect(
      `/edit-text-message-template/${formState.id}`,
      RedirectType.push
    );
  }

  if (data.reviewSMSTemplateAction === 'sms-submit') {
    return redirect(
      `/submit-text-message-template/${formState.id}`,
      RedirectType.push
    );
  }

  throw new Error('Unknown review sms template action.');
}
