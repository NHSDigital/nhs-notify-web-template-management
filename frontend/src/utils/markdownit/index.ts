import MarkdownIt from 'markdown-it';
import { lineBreak } from './plugins/line-break';

export class MarkdownItWrapper extends MarkdownIt {
  constructor() {
    super('zero');

    // rendered links should open in a new tab
    this.renderer.rules.link_open = (tokens, idx) => {
      const href = tokens[idx].attrGet('href');

      return `<a href="${href}" target="_blank" rel="noopener noreferrer">`;
    };
  }

  /**
   * Enables the line break feature
   *
   * @return {this} Returns the instance of the MarkdownItWrapper class for method chaining.
   */
  enableLineBreak() {
    this.use(lineBreak);
    return this;
  }
}

export function renderSMSMarkdown(
  value: string,
  markdown = new MarkdownItWrapper()
) {
  return markdown.render(value);
}

export function renderNHSAppMarkdown(
  value: string,
  markdown = new MarkdownItWrapper()
) {
  markdown.enableLineBreak().enable(['heading', 'link', 'list', 'emphasis']);

  return markdown.render(value);
}

export function renderEmailMarkdown(
  value: string,
  markdown = new MarkdownItWrapper()
) {
  markdown
    .enableLineBreak()
    .enable(['heading', 'link', 'list', 'emphasis', 'hr']);

  return markdown.render(value);
}

export function withEmbeddedLink(
  content: string,
  markdown = new MarkdownItWrapper()
) {
  markdown.enableLineBreak().enable('link');
  return markdown.renderInline(content);
}
