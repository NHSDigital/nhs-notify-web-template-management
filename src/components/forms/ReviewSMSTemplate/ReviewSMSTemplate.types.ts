export type ReviewSMSTemplateProps = {
  templateName: string;
  message: string;
};
