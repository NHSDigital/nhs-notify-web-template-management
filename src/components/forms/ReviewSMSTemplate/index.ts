export * from './ReviewSMSTemplate';
export * from './ReviewSMSTemplate.types';
export * from './server-actions';
