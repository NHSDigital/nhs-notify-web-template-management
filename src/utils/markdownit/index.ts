import MarkdownIt from 'markdown-it';
import { lineBreak } from './plugins/line-break';
import { pageBreak } from './plugins/page-break';

type MarkdownItWrapperProps = {
  enableLineBreaks?: boolean;
  enablePageBreak?: boolean;
};

export const MarkdownItWrapper = (
  opts: MarkdownItWrapperProps = {
    enableLineBreaks: true,
    enablePageBreak: false,
  }
) => {
  const markdownIt = new MarkdownIt('zero');

  if (opts.enableLineBreaks) {
    markdownIt.use(lineBreak);
  }

  if (opts.enablePageBreak) {
    markdownIt.use(pageBreak);
  }

  return markdownIt;
};
