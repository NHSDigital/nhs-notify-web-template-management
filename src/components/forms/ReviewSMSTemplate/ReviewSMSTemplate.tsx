'use client';

import Link from 'next/link';
import { PreviewTemplate } from '@molecules/PreviewTemplate';
import { ReviewTemplate } from '@organisms/ReviewTemplate';
import content from '@content/content';
import { PageComponentProps, SMSTemplate } from '@utils/types';
import { useFormState } from 'react-dom';
import { BackLink } from 'nhsuk-react-components';
import { getBasePath } from '@utils/get-base-path';
import { renderMarkdown, reviewSmsTemplateAction } from './server-actions';

export function ReviewSMSTemplate({
  initialState,
}: Readonly<PageComponentProps<SMSTemplate>>) {
  const {
    components: {
      reviewSMSTemplateContent: { sectionHeading, form },
    },
  } = content;

  const [state, action] = useFormState(reviewSmsTemplateAction, initialState);

  const templateMessage = initialState.message;

  const html = renderMarkdown(templateMessage);

  return (
    <div className='nhsuk-grid-row'>
      <BackLink
        href={`${getBasePath()}/manage-templates`}
        className='nhsuk-u-margin-bottom-5 nhsuk-u-margin-left-3'
      >
        Back to all templates
      </BackLink>
      <ReviewTemplate
        template={initialState}
        sectionHeading={sectionHeading}
        form={{
          ...form,
          state,
          action,
          formId: 'review-sms-template',
          radiosId: 'reviewSMSTemplateAction',
        }}
        PreviewComponent={<PreviewTemplate.Sms message={html} />}
      />
      <p>
        <Link href='/manage-templates'>Back to all templates</Link>
      </p>
    </div>
  );
}
