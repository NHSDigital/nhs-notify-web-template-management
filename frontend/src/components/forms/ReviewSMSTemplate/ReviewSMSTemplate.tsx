'use client';

import Link from 'next/link';
import { PreviewTemplate } from '@molecules/PreviewTemplate';
import { ReviewTemplate } from '@organisms/ReviewTemplate';
import content from '@content/content';
import {
  PageComponentProps,
  SMSTemplate,
} from 'nhs-notify-web-template-management-utils';
import { useFormState } from 'react-dom';
import { BackLink } from 'nhsuk-react-components';
import { getBasePath } from '@utils/get-base-path';
import { renderSMSMarkdown } from '@utils/markdownit';
import { useSearchParams } from 'next/navigation';
import { reviewSmsTemplateAction } from './server-actions';

export function ReviewSMSTemplate({
  initialState,
}: Readonly<PageComponentProps<SMSTemplate>>) {
  const searchParams = useSearchParams();

  const {
    components: {
      reviewSMSTemplateContent: { sectionHeading, form },
    },
  } = content;

  const [state, action] = useFormState(reviewSmsTemplateAction, initialState);
  const templateMessage = initialState.message;
  const html = renderSMSMarkdown(templateMessage);
  const isFromEditPage = searchParams.get('from') === 'edit';

  return (
    <div className='nhsuk-grid-row'>
      <div className='nhsuk-grid-column-full'>
        <BackLink href={`${getBasePath()}/manage-templates`}>
          Back to all templates
        </BackLink>
        <ReviewTemplate
          template={initialState}
          sectionHeading={isFromEditPage ? sectionHeading : undefined}
          form={{
            ...form,
            state,
            action,
            formId: 'review-sms-template',
            radiosId: 'reviewSMSTemplateAction',
          }}
          PreviewComponent={
            <PreviewTemplate.Sms template={initialState} message={html} />
          }
        />
        <p>
          <Link href='/manage-templates'>Back to all templates</Link>
        </p>
      </div>
    </div>
  );
}
