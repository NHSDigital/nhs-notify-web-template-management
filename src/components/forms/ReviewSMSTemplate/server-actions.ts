import { MarkdownItWrapper } from '@utils/markdownit';
import { TemplateFormState } from '@utils/types';
import { redirect, RedirectType } from 'next/navigation';
import { z } from 'zod';

export const $FormSchema = z.object({
  reviewSMSTemplateAction: z.enum(['sms-edit', 'sms-submit'], {
    message: 'Select an option',
  }),
});

export function renderMarkdown(
  value: string,
  markdown = new MarkdownItWrapper()
) {
  return markdown.render(value);
}

export async function reviewSmsTemplateAction(
  formState: TemplateFormState,
  formData: FormData
) {
  const form = Object.fromEntries(formData.entries());

  const { success, error, data } = $FormSchema.safeParse(form);

  if (!success) {
    return {
      ...formState,
      validationError: error.flatten(),
    };
  }

  if (data.reviewSMSTemplateAction === 'sms-edit') {
    return redirect(
      `/create-text-message-template/${formState.id}`,
      RedirectType.push
    );
  }

  if (data.reviewSMSTemplateAction === 'sms-submit') {
    return redirect(
      `/submit-text-message-template/${formState.id}`,
      RedirectType.push
    );
  }

  throw new Error('Unknown review sms template action.');
}
