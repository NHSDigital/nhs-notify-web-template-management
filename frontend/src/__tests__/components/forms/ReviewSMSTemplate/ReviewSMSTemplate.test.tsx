'use client';

import { render, screen } from '@testing-library/react';
import { ReviewSMSTemplate } from '@forms/ReviewSMSTemplate';
import {
  SMSTemplate,
  TemplateFormState,
} from 'nhs-notify-web-template-management-utils';
import { renderSMSMarkdown } from '@utils/markdownit';
import { mockDeep } from 'jest-mock-extended';
import { useSearchParams } from 'next/navigation';

jest.mock('@forms/ReviewSMSTemplate/server-actions');
jest.mock('@utils/markdownit');

jest.mock('react-dom', () => {
  const originalModule = jest.requireActual('react-dom');

  return {
    ...originalModule,
    useFormState: (
      _: (
        formState: TemplateFormState,
        formData: FormData
      ) => Promise<TemplateFormState>,
      initialState: TemplateFormState
    ) => [initialState, '/action'],
  };
});

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(() => ({
    get: jest.fn(() => undefined),
  })),
}));

describe('Review sms form renders', () => {
  it('matches snapshot when navigating from manage templates screen', () => {
    const container = render(
      <ReviewSMSTemplate
        initialState={mockDeep<TemplateFormState<SMSTemplate>>({
          validationError: undefined,
          name: 'test-template-sms',
          message: 'message',
          id: 'template-id',
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when navigating from edit screen', () => {
    const mockSearchParams = new Map([['from', 'edit']]);
    (useSearchParams as jest.Mock).mockImplementation(() => ({
      get: (key: string) => mockSearchParams.get(key),
    }));

    const container = render(
      <ReviewSMSTemplate
        initialState={mockDeep<TemplateFormState<SMSTemplate>>({
          validationError: undefined,
          name: 'test-template-sms',
          message: 'message',
          id: 'template-id',
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('matches error snapshot', () => {
    const container = render(
      <ReviewSMSTemplate
        initialState={mockDeep<TemplateFormState<SMSTemplate>>({
          validationError: {
            formErrors: [],
            fieldErrors: {
              reviewSMSTemplateAction: ['Select an option'],
            },
          },
          name: 'test-template-sms',
          message: 'message',
          id: 'template-id',
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders component correctly', () => {
    render(
      <ReviewSMSTemplate
        initialState={mockDeep<TemplateFormState<SMSTemplate>>({
          validationError: undefined,
          name: 'test-template-sms',
          message: 'message',
          id: 'template-id',
        })}
      />
    );

    expect(screen.getByTestId('sms-edit-radio')).toHaveAttribute(
      'value',
      'sms-edit'
    );

    expect(screen.getByTestId('sms-submit-radio')).toHaveAttribute(
      'value',
      'sms-submit'
    );
  });

  it('should render message with markdown', () => {
    const renderMock = jest.mocked(renderSMSMarkdown);

    renderMock.mockReturnValue('Rendered via MD');

    const message = 'sms message body';

    render(
      <ReviewSMSTemplate
        initialState={mockDeep<TemplateFormState<SMSTemplate>>({
          validationError: undefined,
          name: 'test-template-sms',
          message,
          id: 'template-id',
        })}
      />
    );

    expect(renderMock).toHaveBeenCalledWith(message);

    expect(screen.getByTestId('preview__content-0')).toHaveTextContent(
      'Rendered via MD'
    );
  });
});
