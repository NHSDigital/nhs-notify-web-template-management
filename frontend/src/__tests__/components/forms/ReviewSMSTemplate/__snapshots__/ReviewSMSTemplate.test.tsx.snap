// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review sms form renders matches error snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <div
          class="notify-confirmation-panel nhsuk-heading-l"
        >
          Template saved
        </div>
        <div
          aria-labelledby="error-summary-title"
          class="nhsuk-error-summary"
          role="alert"
          tabindex="-1"
        >
          <h2
            class="nhsuk-error-summary__title"
            data-testid="error-summary"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <ul
            class="nhsuk-list nhsuk-error-summary__list"
          >
            <li>
              <a
                href="#reviewSMSTemplateAction"
              >
                Select an option
              </a>
            </li>
          </ul>
        </div>
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-sms
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              />
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
          </div>
          <div
            class="nhsuk-u-margin-top-4 preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                  data-testid="preview__heading-0"
                  id="preview-heading-message"
                >
                  Message
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="preview__col_content"
                  data-testid="preview__content-0"
                  id="preview-content-message"
                />
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="review-sms-template"
          />
          <div
            class="nhsuk-form-group nhsuk-form-group--error"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="reviewSMSTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group nhsuk-form-group--error"
              >
                <span
                  class="nhsuk-error-message"
                  id="reviewSMSTemplateAction--error-message"
                >
                  <span
                    class="nhsuk-u-visually-hidden"
                  >
                    Error: 
                  </span>
                  Select an option
                </span>
                <div
                  aria-describedby="reviewSMSTemplateAction--error-message"
                  class="nhsuk-radios"
                  id="reviewSMSTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-edit-radio"
                      id="reviewSMSTemplateAction-sms-edit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-edit"
                      id="reviewSMSTemplateAction-sms-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-submit-radio"
                      id="reviewSMSTemplateAction-sms-submit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-submit"
                      id="reviewSMSTemplateAction-sms-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="review-sms-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Review sms form renders matches snapshot when navigating from edit screen 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <div
          class="notify-confirmation-panel nhsuk-heading-l"
        >
          Template saved
        </div>
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-sms
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              />
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
          </div>
          <div
            class="nhsuk-u-margin-top-4 preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                  data-testid="preview__heading-0"
                  id="preview-heading-message"
                >
                  Message
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="preview__col_content"
                  data-testid="preview__content-0"
                  id="preview-content-message"
                />
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="review-sms-template"
          />
          <div
            class="nhsuk-form-group"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="reviewSMSTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group"
              >
                <div
                  class="nhsuk-radios"
                  id="reviewSMSTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-edit-radio"
                      id="reviewSMSTemplateAction-sms-edit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-edit"
                      id="reviewSMSTemplateAction-sms-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-submit-radio"
                      id="reviewSMSTemplateAction-sms-submit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-submit"
                      id="reviewSMSTemplateAction-sms-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="review-sms-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Review sms form renders matches snapshot when navigating from manage templates screen 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-sms
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              />
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
          </div>
          <div
            class="nhsuk-u-margin-top-4 preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                  data-testid="preview__heading-0"
                  id="preview-heading-message"
                >
                  Message
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="preview__col_content"
                  data-testid="preview__content-0"
                  id="preview-content-message"
                />
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="review-sms-template"
          />
          <div
            class="nhsuk-form-group"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="reviewSMSTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group"
              >
                <div
                  class="nhsuk-radios"
                  id="reviewSMSTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-edit-radio"
                      id="reviewSMSTemplateAction-sms-edit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-edit"
                      id="reviewSMSTemplateAction-sms-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="sms-submit-radio"
                      id="reviewSMSTemplateAction-sms-submit"
                      name="reviewSMSTemplateAction"
                      type="radio"
                      value="sms-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="reviewSMSTemplateAction-sms-submit"
                      id="reviewSMSTemplateAction-sms-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="review-sms-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
